### PR TITLE
convert assertion to raised exception in debertav2

### PR DESCRIPTION
**Pull request by sam-h-bean**
# What does this PR do?

Replaces assert with ValueError as per https://github.com/huggingface/transformers/issues/12789.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sgugger

### CODE REVIEW

File: src/transformers/models/deberta/modeling_tf_deberta.py [hunk 1]
```diff
@@ -773,7 +773,8 @@ def call(
         Returns:
             final_embeddings (`tf.Tensor`): output embedding tensor.
         """
-        assert not (input_ids is None and inputs_embeds is None)
+        if input_ids is None and inputs_embeds is None:
+            raise ValueError("input_ids and input_embeds can not both be None")
```
Comment (LysandreJik) [inline src/transformers/models/deberta/modeling_tf_deberta.py:777]: We usually have the same error message everywhere:
```suggestion
            raise ValueError("Need to provide either `input_ids` or `input_embeds`.")
```

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py [hunk 2]
```diff
@@ -870,7 +870,8 @@ def call(
         Returns:
             final_embeddings (`tf.Tensor`): output embedding tensor.
         """
-        assert not (input_ids is None and inputs_embeds is None)
+        if input_ids is None and inputs_embeds is None:
+            raise ValueError("input_ids and input_embeds can not both be None")
```
Comment (LysandreJik) [inline src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py:874]:
```suggestion
            raise ValueError("Need to provide either `input_ids` or `input_embeds`.")
```

**Review by LysandreJik — COMMENTED** [review]
Thank you! Left a small comment :)

File: src/transformers/models/deberta/modeling_tf_deberta.py (hunk 1 above)
Comment (sam-h-bean) [reply src/transformers/models/deberta/modeling_tf_deberta.py:777]: done!

File: src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py (hunk 2 above)
Comment (sam-h-bean) [reply src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py]: done!

**Review by sgugger — APPROVED** [review]
Thanks for your help on this!

**Pull request merged**